packages/volto/src/components/manage/Blocks/Listing/RenderAccordingToContentType.jsx

```diff
@@ -0,0 +1,42 @@
+import config from '@plone/volto/registry';
+import ListPage from '@plone/volto/components/manage/Blocks/Listing/ListPage';
+import ListNews from '@plone/volto/components/manage/Blocks/Listing/ListNewsItem';
+import ListEvent from '@plone/volto/components/manage/Blocks/Listing/ListEventItem';
+
+const contentTemplate = {
+  Document: ListPage,
+  'News Item': ListNews,
+  Event: ListEvent,
+};
+
+const RenderAccordingToContentType = ({
+  item,
+  TitleTag,
+  isEditMode,
+  image,
+  grid,
+}) => {
+  const type = item && item['@type'];
+  const Template =
+    (type &&
+      config?.getComponent &&
+      config.getComponent({ name: 'ListingTemplate', dependencies: [type] })
+        .component) ||
+    contentTemplate[type];
```

Comment:
@iFlameing Using the registry here is good, but I think the way RenderAccordingToContentType is currently implemented, it needs to know too much about the specific listing variations that exist (i.e. the boolean props for image and grid).

Also the contentTemplate mapping should not be hardcoded here. That is what the registry lookup is for.

Instead I would suggest to get rid of RenderAccordingToContentType, and do the registry lookup directly in the variation templates (i.e. in DefaultTemplate, SummaryTemplate, and Grid).

Each variation should use a different component name in the registry lookup:
- default variation -> ListingTemplate component type
- summary variation -> ListingWithImageTemplate component type
- grid variation -> ListingGridTemplate component type

If you have questions about how to use the registry while I am asleep, @sneridagh can probably help.